tools/get_dicom.pl

```diff
@@ -0,0 +1,275 @@
+my @scanTypes    = defined $scanTypes    ? split(',', $scanTypes)    : ();
+
+#---------------------------------------#
+# Read prod file to get DB credentials  #
+#---------------------------------------#
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+if(!@Settings::db) {
+    die "No database settings in file $ENV{LORIS_CONFIG}/.loris_mri/$profile\n";
+}
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+#-------------------------------------------#
+# Fetch tarchiveLibraryDir from the config  #
+#-------------------------------------------#
+my $sth = $dbh->prepare(
+     "SELECT Value "
+   . "FROM Config c "
+   . "JOIN ConfigSettings cs ON (cs.ID=c.ConfigID) "
+   . "WHERE cs.Name = ?"
+);
+$sth->execute('tarchiveLibraryDir');
+
+my($tarchiveLibraryDir) = $sth->fetchrow_array;
+
+#------------------------------------------------------------#
+# Get the list of tarchives for which the patient names      #
+# match the list of supplied patterns                        #
+#------------------------------------------------------------#
+my $stmt = "SELECT c.pscid, s.visit_label, t.dateacquired, t.archivelocation, t.tarchiveid, t.patientname "
+         . "FROM tarchive t "
+         . "JOIN session s ON (t.sessionid=s.id) "
+         . "JOIN candidate c ON (c.candid=s.candid) ";
+if(@patientNames) {
+	my @where = map { "t.patientname LIKE ?" } @patientNames;
+	$stmt .= sprintf("WHERE %s", join("OR", @where));
+}
+$sth = $dbh->prepare($stmt);
+
+if(@patientNames) {
+    for(my $p=0; $p<@patientNames; $p++) {
+        $sth->bind_param($p+1, $patientNames[$p]);
+    }
+}
+$sth->execute();
+
+#----------------------------------------------------------#
+# Create the directory where DICOM files will be extracted #
+#----------------------------------------------------------#
+my $tmpExtractDir = sprintf("%s/%s.%d", $tmpExtractBaseDir, basename($0), $$);
+mkdir $tmpExtractDir or die "Cannot create directory $tmpExtractDir: $!\n";
+
+#---------------------------------------------------------#
+# Process each tarchive and extract the DICOMs associated #
+# to the scan types of interest                           #
+#---------------------------------------------------------#
+my @outDirs = ();
+foreach my $tarchiveRowRef (@{ $sth->fetchall_arrayref }) {
+    my($pscid, $visitLabel, $dateAcquired, $archiveLocation, $tarchiveId) = @$tarchiveRowRef;
+    
+    my($innerTar) = "$archiveLocation.gz" =~ /_(ImagingUpload.*.tar.gz)$/;
+    
+    # Extract only the .tar.gz archive from the main archive (ignore the 
+    # meta data and log file)
+    print "Extracting $innerTar in $tmpExtractDir...";
+    system("tar xf $tarchiveLibraryDir/$archiveLocation -C $tmpExtractDir $innerTar") == 0 
+      or die "Extraction of $innerTar from $tarchiveLibraryDir/$archiveLocation in directory $tmpExtractDir failed: $?";
+    print "done\n";
+    
+    # Fetch all the MINC files in the archive whose acquisition protocols
```

Comment:
Fetch all the MINC files **created out of the DICOM archive** whose acquisition protocols